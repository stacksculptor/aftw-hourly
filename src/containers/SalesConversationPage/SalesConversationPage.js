import React from 'react';
import { Link } from 'react-router';
import { Page } from '../../components';

export default (props) => {
  const { params } = props;
  return (
    <Page title="Sales conversation page" >
      <p>Sale id: { params.id }</p>
      <Link to={ `/sale/${params.id}/discussion` }>Discussion tab</Link>
      <br />
      <Link to={ `/sale/${params.id}/details` }>Details tab</Link>
      <p>Mobile layout needs different views for discussion and details.</p>
      <p>Discussion view is the default if route doesn't specify mobile tab (e.g. <i>/order/1234</i>)</p>
    </Page>
  );
};
