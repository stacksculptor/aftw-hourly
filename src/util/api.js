import { types as sdkTypes, transit } from './sdkLoader';
import Decimal from 'decimal.js';

const apiBaseUrl = () => {
  const port = process.env.REACT_APP_DEV_API_SERVER_PORT;
  const useDevApiServer = process.env.NODE_ENV === 'development' && !!port;

  // In development, the dev API server is running in a different port
  if (useDevApiServer) {
    return `http://localhost:${port}`;
  }

  // Otherwise, use the same domain and port as the frontend
  return '/';
};

const typeHandlers = [
  {
    type: sdkTypes.BigDecimal,
    customType: Decimal,
    writer: v => new sdkTypes.BigDecimal(v.toString()),
    reader: v => new Decimal(v.value),
  },
];

const serialize = data => {
  return transit.write(data, { typeHandlers });
};

const deserialize = str => {
  return transit.read(str, { typeHandlers });
};

const post = (path, body) => {
  const url = `${apiBaseUrl()}${path}`;
  const options = {
    method: 'POST',
    credentials: 'include',
    headers: {
      'Content-Type': 'application/transit+json',
    },
    body: serialize(body),
  };
  return window
    .fetch(url, options)
    .then(res => {
      if (res.status >= 400) {
        const e = new Error('Local API request failed');
        e.apiResponse = res;
        throw e;
      }
      return res;
    })
    .then(res => {
      const contentTypeHeader = res.headers.get('Content-Type');
      const contentType = contentTypeHeader ? contentTypeHeader.split(';')[0] : null;
      if (contentType === 'application/transit+json') {
        return res.text().then(deserialize);
      } else if (contentType === 'application/json') {
        return res.json();
      }
      return res.text();
    });
};

export const transactionLineItems = body => {
  return post('/api/transaction-line-items', body);
};

export const initiatePrivileged = body => {
  return post('/api/initiate-privileged', body);
};
