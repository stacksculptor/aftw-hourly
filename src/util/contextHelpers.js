import React, { PropTypes, Component as ReactComponent } from 'react'; // eslint-disable-line react/no-deprecated
import { throttle } from 'lodash';

/**
 * A higher order component (HOC) to take the togglePageClassNames function from
 * the context that the Page component has provided.
 */
export const withTogglePageClassNames = Component => {
  const WithTogglePageClassNamesComponent = (props, context) => (
    <Component togglePageClassNames={context.togglePageClassNames} {...props} />
  );

  WithTogglePageClassNamesComponent.displayName = `withTogglePageClassNames(${Component.displayName || Component.name})`;

  const { func } = PropTypes;

  WithTogglePageClassNamesComponent.contextTypes = {
    togglePageClassNames: func.isRequired,
  };

  return WithTogglePageClassNamesComponent;
};

/**
 * A higher order component (HOC) that provides the current viewport
 * dimensions to the wrapped component as a `viewport` prop that has
 * the shape `{ width: 600, height: 400}`.
 */
export const withViewport = Component => {
  // The resize event is flooded when the browser is resized. We'll
  // use a small timeout to throttle changing the viewport since it
  // will trigger rerendering.
  const WAIT_MS = 100;

  class WithViewportComponent extends ReactComponent {
    constructor(props) {
      super(props);
      this.state = { width: 0, height: 0 };
      this.handleWindowResize = this.handleWindowResize.bind(this);
      this.setViewport = throttle(this.setViewport.bind(this), WAIT_MS);
    }
    componentDidMount() {
      this.setViewport();
      window.addEventListener('resize', this.handleWindowResize);
    }
    componentWillUnmount() {
      window.removeEventListener('resize', this.handleWindowResize);
    }
    handleWindowResize() {
      this.setViewport();
    }
    setViewport() {
      this.setState({
        width: window.innerWidth,
        height: window.innerHeight,
      });
    }
    render() {
      const viewport = this.state;
      const props = { ...this.props, viewport };
      return <Component {...props} />;
    }
  }

  WithViewportComponent.displayName = `withViewport(${Component.displayName || Component.name})`;

  return WithViewportComponent;
};
