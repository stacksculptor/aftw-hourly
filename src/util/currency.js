import { trimEnd } from 'lodash';
import Decimal from 'decimal.js';

// This works for USD and EUR
const SUB_UNITS_DIVISOR = 100;

// Default config for currency formatting (for React-Intl).
export const currencyDefaultConfig = {
  style: 'currency',
  currency: 'USD',
  currencyDisplay: 'symbol',
  useGrouping: true,
  minimumFractionDigits: 2,
  maximumFractionDigits: 2,
};

// Ensures that the given string uses only dots or points
// ensureSeparator('9999999,99', false) // => '9999999.99'
export const ensureSeparator = (str, usePoint = false) => {
  if (typeof str !== 'string') {
    throw new Error('Parameter must be a string');
  }
  return usePoint ? str.split('.').join(',') : str.split(',').join('.');
};

// Ensures that the given string uses only dots (e.g. JavaScript floats use dots)
export const ensureDotSeparator = str => {
  return ensureSeparator(str, false);
};

// Convert string to Decimal object (from Decimal.js math library)
export const convertToDecimal = str => {
  const dotFormattedStr = ensureDotSeparator(str);
  try {
    return new Decimal(dotFormattedStr);
  } catch (e) {
    throw e;
  }
};

// Converts Decimal object to string
export const convertDecimalToString = (decimalValue, usePoint = false) => {
  try {
    const d = new Decimal(decimalValue);
    return ensureSeparator(d.toString(), usePoint);
  } catch (e) {
    throw e;
  }
};

// Limits value to sub-unit precision: "1.4567" -> "1.45"
// Useful in input fields so this doesn't use rounding.
export const truncateToSubUnitPrecision = (inputString, currency, usePoint = false) => {
  if (!(currency === 'USD' || currency === 'EUR')) {
    // Other currencies sub unit support need to be changed case by case
    throw new Error('Currency must be either USD or EUR');
  }

  // '10,' should be passed through, but that format is not supported as valid number
  const trimmed = trimEnd(inputString, usePoint ? ',' : '.');
  // create another instance and check if value is convertable
  const value = convertToDecimal(trimmed, usePoint);

  if (value.isNegative()) {
    throw new Error('Parameter must be a positive number');
  }

  // Amount is always counted in subunits
  // E.g. $10 => 1000¢
  const amount = value.times(SUB_UNITS_DIVISOR);

  // Amount must be integer
  // We don't deal with subunit fragments like 1000.345¢
  if (amount.isInteger()) {
    // accepted strings: '9', '9,' '9.' '9,99'
    const decimalCount2 = value.toFixed(2);
    const decimalPrecisionMax2 = decimalCount2.length >= inputString.length
      ? inputString
      : value.toFixed(2);
    return ensureSeparator(decimalPrecisionMax2, usePoint);
  } else {
    // truncate strings ('9.999' => '9.99')
    const truncated = amount.truncated().dividedBy(SUB_UNITS_DIVISOR);
    return convertDecimalToString(truncated, usePoint);
  }
};

// Converts given value to sub unit value and returns it as a number
export const convertUnitToSubUnit = (value, currency, usePoint = false) => {
  if (!(currency === 'USD' || currency === 'EUR')) {
    throw new Error('Currency must be either USD or EUR');
  }

  if (!(typeof value === 'string' || typeof value === 'number')) {
    throw new Error('Value must be either number or string');
  }

  const val = typeof value === 'string' ? convertToDecimal(value, usePoint) : new Decimal(value);
  const amount = val.times(SUB_UNITS_DIVISOR);

  if (amount.isInteger()) {
    return amount.toNumber();
  } else {
    throw new Error(`value must divisible by ${SUB_UNITS_DIVISOR}`);
  }
};
