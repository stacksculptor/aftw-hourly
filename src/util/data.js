import { reduce } from 'lodash';

/**
 * Combine the given relationships objects
 *
 * See: http://jsonapi.org/format/#document-resource-object-relationships
 */
export const combinedRelationships = (oldRels, newRels) => {
  if (!oldRels && !newRels) {
    // Special case to avoid adding an empty relationships object when
    // none of the resource objects had any relationships.
    return null;
  }
  return { ...oldRels, ...newRels };
};

/**
 * Combine the given resource objects
 *
 * See: http://jsonapi.org/format/#document-resource-objects
 */
export const combinedResourceObjects = (oldRes, newRes) => {
  const { id, type } = oldRes;
  if (newRes.id.uuid !== id.uuid || newRes.type !== type) {
    throw new Error('Cannot merge resource objects with different ids or types');
  }
  const attributes = newRes.attributes || oldRes.attributes;
  const attrs = attributes ? { attributes } : null;
  const relationships = combinedRelationships(oldRes.relationships, newRes.relationships);
  const rels = relationships ? { relationships } : null;
  return { id, type, ...attrs, ...rels };
};

/**
 * Combine the resource objects form the given api response to the
 * existing entities.
 */
export const updatedEntities = (oldEntities, apiResponse) => {
  const { data, included = [] } = apiResponse;
  const objects = (Array.isArray(data) ? data : [data]).concat(included);

  /* eslint-disable no-param-reassign */
  const newEntities = objects.reduce(
    (entities, curr) => {
      const { id, type } = curr;
      entities[type] = entities[type] || {};
      const entity = entities[type][id.uuid];
      entities[type][id.uuid] = entity ? combinedResourceObjects(entity, curr) : curr;
      return entities;
    },
    oldEntities,
  );
  /* eslint-enable no-param-reassign */

  return newEntities;
};

/**
 * Denormalise the entities with the given IDs from the entities object
 *
 * This function calculates the dernormalised tree structure from the
 * normalised entities object with all the relationships joined in.
 *
 * @param {Object} entities entities object in the SDK Redux store
 * @param {String} type entity type of the given IDs
 * @param {Array<UUID>} ids IDs to pick from the entities
 */
export const denormalisedEntities = (entities, type, ids) => {
  if (!entities[type] && ids.length > 0) {
    throw new Error(`No entities of type ${type}`);
  }
  return ids.map(id => {
    const entity = entities[type][id.uuid];
    if (!entity) {
      throw new Error(`Entity ${type} with id ${id.uuid} not found`);
    }
    const { relationships, ...entityData } = entity;

    if (relationships) {
      // Recursively join in all the relationship entities
      return reduce(
        relationships,
        (ent, relRef, relName) => {
          // A relationship reference can be either a single object or
          // an array of objects. We want to keep that form in the final
          // result.
          const hasMultipleRefs = Array.isArray(relRef.data);
          const refs = hasMultipleRefs ? relRef.data : [relRef.data];
          const relIds = refs.map(ref => ref.id);
          const relType = refs[0].type;
          const rels = denormalisedEntities(entities, relType, relIds);

          // eslint-disable-next-line no-param-reassign
          ent[relName] = hasMultipleRefs ? rels : rels[0];
          return ent;
        },
        entityData,
      );
    }
    return entityData;
  });
};
