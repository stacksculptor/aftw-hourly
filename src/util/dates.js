import moment from 'moment';

/**
 * Input names for the DateRangePicker from react-dates.
 */
export const START_DATE = 'startDate';
export const END_DATE = 'endDate';

/**
 * Check that the given parameter is a Date object.
 *
 * @param {Date} object that should be a Date.
 *
 * @returns {boolean} true if given parameter is a Date object.
 */
export const isDate = d =>
  d && Object.prototype.toString.call(d) === '[object Date]' && !Number.isNaN(d.getTime());

/**
 * Check if the given parameters represent the same Date value (timestamps are compared)
 *
 * @param {Date} first param that should be a Date and it should have same timestamp as second param.
 * @param {Date} second param that should be a Date and it should have same timestamp as second param.
 *
 * @returns {boolean} true if given parameters have the same timestamp.
 */
export const isSameDate = (a, b) => a && isDate(a) && b && isDate(b) && a.getTime() === b.getTime();

/**
 * Convert date given by API to something meaningful noon on browser's timezone
 * So, what happens is that date given by client
 * ("Fri Mar 30 2018 12:00:00 GMT-1100 (SST)" aka "Fri Mar 30 2018 23:00:00 GMT+0000 (UTC)")
 * will be read as UTC time. Then API normalizes night/day bookings to
 * start from 00:00 UTC (i.e. discards hours from UTC day).
 * So Api gives 00:00 UTC which (in our example) would be locally
 * "Thu Mar 29 2018 13:00:00 GMT-1100 (SST)".
 *
 * The resulting timestamp from API is:
 * localTimestamp.subtract(12h).add(timezoneoffset) (in eg. -23 h)
 *
 * So, this function adds those removed hours back.
 *
 * @param {Date} date is a local date object
 *
 * @returns {Date} date (given by API as UTC 00:00) converted back to local noon.
 */
export const dateFromAPIToLocalNoon = date => {
  const timezoneDiffInMinutes = moment(date).utcOffset();
  // Example timezone SST:
  // We get a Fri 00:00 UTC aka "Thu Mar 29 2018 13:00:00 GMT-1100 (SST)"
  // We need to subtract timezone difference (-11h), effectively adding 11h - to get to correct date
  const momentInLocalTimezone = moment(date).subtract(timezoneDiffInMinutes, 'minutes');
  // To be on the safe zone with leap seconds and stuff when using day / night picker
  // we'll add 12 h to get to the noon of day in local timezone.
  return momentInLocalTimezone.add(12, 'hours').toDate();
};

/**
 * Convert local date for API.
 * Date given by browser
 * ("Fri Mar 30 2018 12:00:00 GMT-1100 (SST)" aka "Fri Mar 30 2018 23:00:00 GMT+0000 (UTC)")
 * must be modified so that API will get correct moment also in UTC.
 * We achieve this by adding timezone offset to local date / timestamp.
 *
 * The resulting timestamp for the API is:
 * localTimestamp.add(timezoneoffset)
 * In eg. Fri Mar 30 2018 23:00:00 GMT-1100 (SST) aka "Fri Mar 30 2018 12:00:00 GMT+0000 (UTC)"
 *
 * @param {Date} date is a local date object
 *
 * @returns {Date} date (given by API as UTC 00:00) converted back to local noon.
 */
export const dateFromLocalToAPI = date => {
  const timezoneDiffInMinutes = moment(date).utcOffset();
  const momentInLocalTimezone = moment(date).add(timezoneDiffInMinutes, 'minutes');

  return momentInLocalTimezone.toDate();
};

/**
 * Calculate the number of nights between the given dates
 *
 * @param {Date} startDate start of the time period
 * @param {Date} endDate end of the time period
 *
 * @throws Will throw if the end date is before the start date
 * @returns {Number} number of nights between the given dates
 */
export const nightsBetween = (startDate, endDate) => {
  const nights = moment(endDate).diff(startDate, 'days');
  if (nights < 0) {
    throw new Error('End date cannot be before start date');
  }
  return nights;
};

/**
 * Calculate the number of days between the given dates
 *
 * @param {Date} startDate start of the time period
 * @param {Date} endDate end of the time period. NOTE: with daily
 * bookings, it is expected that this date is the exclusive end date,
 * i.e. the last day of the booking is the previous date of this end
 * date.
 *
 * @throws Will throw if the end date is before the start date
 * @returns {Number} number of days between the given dates
 */
export const daysBetween = (startDate, endDate) => {
  const days = moment(endDate).diff(startDate, 'days');
  if (days < 0) {
    throw new Error('End date cannot be before start date');
  }
  return days;
};

/**
 * Format the given date
 *
 * @param {Date} date to be formatted
 *
 * @returns {String} formatted month string
 */
export const monthIdString = date => moment(date).format('YYYY-MM');

/**
 * Format the given date
 *
 * @param {Object} intl Intl object from react-intl
 * @param {String} todayString translation for the current day
 * @param {Date} d Date to be formatted
 *
 * @returns {String} formatted date
 */
export const formatDate = (intl, todayString, d) => {
  const paramsValid = intl && d instanceof Date && typeof todayString === 'string';
  if (!paramsValid) {
    throw new Error(`Invalid params for formatDate: (${intl}, ${todayString}, ${d})`);
  }
  const now = moment(intl.now());
  const formattedTime = intl.formatTime(d);
  let formattedDate;

  if (now.isSame(d, 'day')) {
    // e.g. "Today, 9:10pm"
    formattedDate = todayString;
  } else if (now.isSame(d, 'week')) {
    // e.g. "Wed, 8:00pm"
    formattedDate = intl.formatDate(d, {
      weekday: 'short',
    });
  } else if (now.isSame(d, 'year')) {
    // e.g. "Aug 22, 7:40pm"
    formattedDate = intl.formatDate(d, {
      month: 'short',
      day: 'numeric',
    });
  } else {
    // e.g. "Jul 17 2016, 6:02pm"
    const date = intl.formatDate(d, {
      month: 'short',
      day: 'numeric',
    });
    const year = intl.formatDate(d, {
      year: 'numeric',
    });
    formattedDate = `${date} ${year}`;
  }

  return `${formattedDate}, ${formattedTime}`;
};
