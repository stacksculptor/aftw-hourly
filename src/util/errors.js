/**
 * ================ API error handling utilities ================
 *
 * This module exports helpers that can be used to check if API
 * response errors are some specific error cases.
 *
 * NOTE: most of the functions are tied to an endpoint, and should not
 * be used to check error responses from any other endpoint. Check the
 * name and the docstring of the function to ensure correct usage.
 */

const ERROR_CODE_TRANSITION_PARAMETER_VALIDATION_FAILED = 'transition-parameter-validation-failed';
const ERROR_CODE_PAYMENT_FAILED = 'transaction-payment-failed';
const ERROR_CODE_EMAIL_TAKEN = 'email-taken';
const ERROR_CODE_TOO_MANY_VERIFICATION_REQUESTS = 'email-too-many-verification-requests';
const ERROR_CODE_UPLOAD_OVER_LIMIT = 'request-upload-over-limit';

const responseErrors = apiError => {
  return apiError && apiError.data && apiError.data.errors ? apiError.data.errors : [];
};

const hasErrorWithCode = (apiError, code) => {
  return responseErrors(apiError).some(error => {
    return error.code === code;
  });
};

/**
 * Check if the given API error (from `sdk.currentuser.create()`) is
 * due to the email address already being in use.
 */
export const isSignupEmailTakenError = error => hasErrorWithCode(error, ERROR_CODE_EMAIL_TAKEN);

/**
 * Check if the given API error (from `sdk.currentuser.changeEmail()`) is
 * due to the email address already being in use.
 */
export const isChangeEmailTakenError = error => hasErrorWithCode(error, ERROR_CODE_EMAIL_TAKEN);

/**
 * Check if the given API error (from
 * `sdk.currentUser.sendVerificationEmail()`) is due to too many
 * active email verification requests.
 *
 * There qre only a specific amount of active verification requests
 * allowed, and the user has to wait for them to expire to be able to
 * request sending new verification emails.
 */
export const isTooManyEmailVerificationRequestsError = error =>
  hasErrorWithCode(error, ERROR_CODE_TOO_MANY_VERIFICATION_REQUESTS);

/**
 * Check if the given API error (from
 * `sdk.images.uploadListingImage()`) is due to the image being over
 * the size limit.
 */
export const isUploadListingImageOverLimitError = error =>
  hasErrorWithCode(error, ERROR_CODE_UPLOAD_OVER_LIMIT);

/**
 * Check if the given API error (from
 * `sdk.images.uploadProfileImage()`) is due to the image being over
 * the size limit.
 */
export const isUploadProfileImageOverLimitError = error =>
  hasErrorWithCode(error, ERROR_CODE_UPLOAD_OVER_LIMIT);

/**
 * Check if the given API error (from `sdk.passwordReset.request()`)
 * is due to no user having the given email address.
 */
export const isPasswordRecoveryEmailNotFoundError = error => error && error.status === 404;

/**
 * Check if the given API error (from `sdk.passwordReset.request()`)
 * is due to the email not being verified, preventing the reset.
 */
export const isPasswordRecoveryEmailNotVerifiedError = error => error && error.status === 409;

/**
 * Check if the given API error (from `sdk.transaction.initiate()` or
 * `sdk.transaction.initiateSpeculative()`) is due to the listing
 * being closed or deleted.
 */
export const isTransactionInitiateListingNotFoundError = error => {
  return responseErrors(error).some(apiError => {
    let notfound = false;

    try {
      // TODO: This is clearly a temporary solution for digging out
      // the deleted information from the error and should be changed
      // to a proper error code when the API supports a specific code
      // for this.
      notfound = apiError.details.data.rep[0][1].reason === 'listing-not-found';
    } catch (e) {
      // Ignore
    }

    return apiError.code === ERROR_CODE_TRANSITION_PARAMETER_VALIDATION_FAILED && notfound;
  });
};

/**
 * Check if the given API error (from `sdk.transaction.initiate()`) is
 * due to the transaction total amount being too low for Stripe.
 */
export const isTransactionInitiateAmountTooLowError = error => {
  return responseErrors(error).some(apiError => {
    const isPaymentFailedError =
      apiError.status === 402 && apiError.code === ERROR_CODE_PAYMENT_FAILED;
    let isAmountTooLow = false;

    try {
      // TODO: This is a temporary solution until a proper error code
      // for this specific error is received in the response.
      const msg = apiError.details.msg;
      isAmountTooLow = msg.startsWith('Amount must be at least');
    } catch (e) {
      // Ignore
    }

    return isPaymentFailedError && isAmountTooLow;
  });
};

/**
 * Check if the given API error (from `sdk.currentUser.changeEmail(params)`)
 * is due to giving wrong password.
 */
export const isChangeEmailWrongPassword = error => error && error.status === 403;

/**
 * Check if the given API error (from `sdk.currentUser.changePassword(params)`)
 * is due to giving wrong password.
 */
export const isChangePasswordWrongPassword = error => error && error.status === 403;
