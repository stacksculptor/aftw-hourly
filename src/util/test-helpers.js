import React from 'react';
import { IntlProvider } from 'react-intl';
import { BrowserRouter } from 'react-router';
import { Provider } from 'react-redux';
import configureStore from '../store';

export const TestProvider = props => {
  const store = configureStore();
  return (
    <IntlProvider locale="en">
      <BrowserRouter>
        <Provider store={store}>
          {props.children}
        </Provider>
      </BrowserRouter>
    </IntlProvider>
  );
};
