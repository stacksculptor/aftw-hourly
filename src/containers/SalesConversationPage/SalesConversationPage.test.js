import React from 'react';
import { renderTree } from '../../util/test-helpers';
import SalesConversationPage from './SalesConversationPage';

describe('SalesConversationPage', () => {
  it('matches snapshot', () => {
    const tree = renderTree(<SalesConversationPage params={{ id: 12345 }} />);
    expect(tree).toMatchSnapshot();
  });
});
