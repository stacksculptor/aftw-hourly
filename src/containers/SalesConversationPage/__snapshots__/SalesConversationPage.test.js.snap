exports[`SalesConversationPage matches snapshot 1`] = `
<div
  className="">
  <div
    className={undefined}>
    <div>
      <a
        className=""
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#127968;",
          }
        }
        href="/"
        onClick={[Function]}
        style={Object {}} />
      <select
        className={undefined}
        onChange={[Function]}
        value="default">
        <option
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#127828;",
            }
          }
          value="default" />
        <option
          value="/s">
          Search
        </option>
        <option
          value="/l/Bike-Pelago/12345">
          Listing page
        </option>
        <option
          value="/u/Bikerrs">
          Profile
        </option>
        <option
          value="/u/Bikerrs/edit">
          Edit profile
        </option>
        <option
          value="/checkout/lid1234">
          Checkout
        </option>
        <option
          value="/inbox">
          Inbox
        </option>
        <option
          value="/orders">
          Orders
        </option>
        <option
          value="/sales">
          Sales
        </option>
        <option
          value="/password/forgotten">
          Request password
        </option>
        <option
          value="/password/change">
          Change password
        </option>
        <option
          value="/listings">
          Manage listings
        </option>
        <option
          value="/account">
          Account settings
        </option>
        <option
          value="/account/contact-details">
          Contact details
        </option>
        <option
          value="/account/payout-preferences">
          Payout preferences
        </option>
        <option
          value="/account/security">
          Security
        </option>
      </select>
    </div>
    <div
      className={undefined}>
      <a
        className=""
        href="/login"
        onClick={[Function]}
        style={Object {}}>
        Login
      </a>
    </div>
  </div>
  <h1>
    Sales conversation page
  </h1>
  <p>
    Sale id: 
    12345
  </p>
  <a
    className=""
    href="/sale/12345/discussion"
    onClick={[Function]}
    style={Object {}}>
    Discussion tab
  </a>
  <br />
  <a
    className=""
    href="/sale/12345/details"
    onClick={[Function]}
    style={Object {}}>
    Details tab
  </a>
  <p>
    Mobile layout needs different views for discussion and details.
  </p>
  <p>
    Discussion view is the default if route doesn\'t specify mobile tab (e.g. 
    <i>
      /order/1234
    </i>
    )
  </p>
</div>
`;
